sinks/api/transport.go

```diff
@@ -7,30 +7,81 @@ package api
 import (
 	"context"
 	"encoding/json"
+	kitot "github.com/go-kit/kit/tracing/opentracing"
 	kithttp "github.com/go-kit/kit/transport/http"
 	"github.com/go-zoo/bone"
 	"github.com/ns1labs/orb"
+	"github.com/ns1labs/orb/internal/httputil"
 	"github.com/ns1labs/orb/pkg/db"
 	"github.com/ns1labs/orb/pkg/errors"
 	"github.com/ns1labs/orb/pkg/types"
 	"github.com/ns1labs/orb/sinks"
+	"github.com/opentracing/opentracing-go"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 	"io"
 	"net/http"
 	"strings"
 )
 
-func MakeHandler(svcName string, svc sinks.Service) http.Handler {
+const (
+	offsetKey   = "offset"
+	limitKey    = "limit"
+	nameKey     = "name"
+	orderKey    = "order"
+	dirKey      = "dir"
+	metadataKey = "metadata"
+	defOffset   = 0
+	defLimit    = 10
+)
+
+func MakeHandler(tracer opentracing.Tracer, svcName string, svc sinks.Service) http.Handler {
 	opts := []kithttp.ServerOption{
 		kithttp.ServerErrorEncoder(encodeError),
 	}
 	r := bone.New()
 
 	r.Post("/sinks", kithttp.NewServer(
-		addEndpoint(svc),
+		kitot.TraceServer(tracer, "create_sink")(addEndpoint(svc)),
 		decodeAddRequest,
 		types.EncodeResponse,
-		opts...))
+		opts...,
+	))
+	r.Put("/sinks/:id", kithttp.NewServer(
+		kitot.TraceServer(tracer, "edit_sink")(updateSinkEndpoint(svc)),
+		decodeEditRequest,
+		types.EncodeResponse,
+		opts...,
+	))
+	r.Get("/sinks", kithttp.NewServer(
+		kitot.TraceServer(tracer, "list_sinks")(listSinksEndpoint(svc)),
+		decodeList,
+		types.EncodeResponse,
+		opts...,
+	))
+	r.Get("/features/sinks", kithttp.NewServer(
+		kitot.TraceServer(tracer, "list_backends")(listBackendsEndpoint(svc)),
+		decodeListBackends,
+		types.EncodeResponse,
+		opts...,
+	))
+	r.Get("/features/sinks/:id", kithttp.NewServer(
```

Comment:
in the original openapi spec this endpoint did not exist, it listed all information in the `/features/sinks` endpoint. I am fine to do it this way but we would need to update the openapi spec to match